filters/openpolicyagent/openpolicyagent.go

```diff
@@ -409,9 +417,22 @@ func (opa *OpenPolicyAgentInstance) waitPluginsReady(checkInterval, timeout time
 		return true
 	}
 
-	opa.Logger().Debug("Waiting for plugins activation (%v).", timeout)
+	err := opautil.WaitFunc(pluginsReady, checkInterval, timeout)
+
+	if err != nil {
+		for pluginName, status := range opa.manager.PluginStatus() {
+			if status != nil && status.State != plugins.StateOK {
+				opa.Logger().WithFields(map[string]interface{}{
+					"plugin_name":   pluginName,
+					"plugin_state":  status.State,
+					"error_message": status.Message,
+				}).Error("Open policy agent plugin did not start in %v", timeout)
+			}
+		}
+		err = fmt.Errorf("one or more open policy agent plugins failed to start in %v with error: %v", timeout, err.Error())
```

Comment:
```suggestion
		err = fmt.Errorf("one or more open policy agent plugins failed to start in %v with error: %w", timeout, err)
```
This allows error chaining.